beanie/odm/fields.py

```diff
@@ -14,14 +14,19 @@
 )
 
 
-def Indexed(typ, index_type=ASCENDING):
+def Indexed(typ, index_type=ASCENDING, unique=False):
     """
-    Returns a subclass of `typ` with an extra attribute `_indexed` et to True.
+    Returns a subclass of `typ` with an extra attribute `_indexed` as a tuple:
+    - Index 0 index type set to `pymongo.ASCENDING` or `pymongo.DESCENDING`
+    - Index 1 index unique set to False or True
     When instantiated the type of the result will actually be `typ`.
     """
 
+    if index_type not in (1, -1):
```

Comment:
There are some other index types, like Text indexes or Geospatial. And other types could be added later. It is better to let the driver check this and not do this on the ODM level.

Reply:
Removed check in latest commit